configs/testing-library-compass/src/index.tsx

```diff
@@ -366,156 +390,46 @@ function renderWithConnections<
-  {
-    initialProps,
-    ...options
-  }: RenderConnectionsOptions<C, BE> & {
-    initialProps?: HookProps;
-  } = {}
-) {
-  const result = { current: null } as { current: HookResult };
-  function HookResultGetter(props: HookProps) {
-    result.current = cb(props);
-    return null;
-  }
-  const { result: renderResult, ...rest } = renderWithConnections(
-    <HookResultGetter {...(initialProps as any)}></HookResultGetter>,
-    options
-  );
-  return {
-    ...rest,
-    rerender: (props?: HookProps) => {
-      return renderResult.rerender(
-        <HookResultGetter {...(props as any)}></HookResultGetter>
-      );
-    },
-    result,
-  };
-}
+export type RenderHookConnectionsOptions<HookProps> = Omit<
+  RenderHookOptions<HookProps>,
+  'wrapper'
+> & { wrapper?: ComponentWithChildren } & TestConnectionsOptions;
 
-function renderPluginComponentWithConnections<
-  T,
-  S extends Record<string, () => unknown>,
-  A extends HadronPlugin,
-  C extends Element | DocumentFragment = HTMLElement,
-  BE extends Element | DocumentFragment = C
->(
-  ui: React.ReactElement,
-  Plugin: HadronPluginComponent<T, S, A>,
-  initialProps: T,
-  options: RenderConnectionsOptions<C, BE> = {}
-) {
-  let plugin;
-  function ComponentWithProvider() {
-    plugin = Plugin.useActivate(initialProps);
-    return (
-      <Provider store={plugin.store} context={plugin.context}>
-        {ui}
-      </Provider>
-    );
-  }
-  const result = renderWithConnections(
-    <ComponentWithProvider></ComponentWithProvider>,
-    options
-  );
-  return {
-    plugin: plugin as unknown as A,
-    ...result,
-  };
-}
+export type RenderWithConnectionsHookResult<
+  HookProps = unknown,
+  HookResult = unknown
+> = ReturnType<typeof createWrapper>['wrapperState'] &
+  RenderHookResult<HookProps, HookResult>;
 
-function renderPluginHookWithConnections<
-  HookResult,
-  T,
-  S extends Record<string, () => unknown>,
-  A extends HadronPlugin,
-  C extends Element | DocumentFragment = HTMLElement,
-  BE extends Element | DocumentFragment = C
->(
-  cb: () => HookResult,
-  Plugin: HadronPluginComponent<T, S, A>,
-  initialProps: T,
-  options: RenderConnectionsOptions<C, BE> = {}
-) {
-  const result = { current: null } as { current: HookResult };
-  function HookResultGetter() {
-    result.current = cb();
-    return null;
-  }
-  const {
-    // eslint-disable-next-line @typescript-eslint/no-unused-vars
-    result: _renderResult,
-    ...rest
-  } = renderPluginComponentWithConnections(
-    <HookResultGetter></HookResultGetter>,
-    Plugin,
+function renderHookWithConnections<HookProps, HookResult>(
+  cb: (props: HookProps) => HookResult,
+  {
     initialProps,
-    options
+    wrapper,
+    ...connectionsOptions
+  }: RenderHookConnectionsOptions<HookProps> = {}
+): RenderWithConnectionsHookResult<HookProps, HookResult> {
+  const { wrapper: Wrapper, wrapperState } = createWrapper(
+    connectionsOptions,
+    true,
+    wrapper as ComponentWithChildren
```

Comment:
Maybe I'm missing something, but

```suggestion
    wrapper
```

compiles fine for me

Reply:
Thanks for spotting! Yeah, I cleaned it up and removed the assertion from most of the places, but somehow missed this one